sound/soc/sof/ipc4-topology.h

```diff
@@ -222,6 +222,44 @@ struct sof_ipc4_dma_stream_ch_map {
 	struct sof_ipc4_dma_device_stream_ch_map mapping[SOF_IPC4_DMA_DEVICE_MAX_COUNT];
 } __packed;
 
+#define SOF_IPC4_DMA_METHOD_HDA   1
+#define SOF_IPC4_DMA_METHOD_GPDMA 2
+
+/**
+ * struct sof_ipc4_dma_config: DMA configuration
+ * @dma_method: HDAudio or GPDMA
+ * @pre_allocated_by_host: 1 if host driver allocates DMA channels, 0 otherwise
+ * @dma_channel_id: for HDaudio defined as @stream_id - 1
+ * @stream_id: HDaudio stream tag
+ * @dma_stream_channel_map: array of device/channel mappings
+ * @dma_priv_config_size: currently not used
+ * @dma_priv_config: currently not used
+ */
+struct sof_ipc4_dma_config {
+	uint8_t dma_method;
+	uint8_t pre_allocated_by_host;
+	uint16_t rsvd;
+	uint32_t dma_channel_id;
+	uint32_t stream_id;
+	struct sof_ipc4_dma_stream_ch_map dma_stream_channel_map;
+	uint32_t  dma_priv_config_size;
+	uint8_t dma_priv_config[0];
+} __packed;
```

Comment:
do we need to align 4 ?

Reply:
that's a good question. I use __packed because that's what was used before.

I would guess the memcpy does remove any need for alignment, no? The structure will be copied over anyways, so what we want to make sure of is that there's no holes and a minimal size.

@RanderWang @bardliao @jxstelter @ranj063 please check if this makes any sense to you!